### PR TITLE
ref(replay): Refactor Replay Details>Network table to use new *Frame types

### PR DESCRIPTION
**Pull request by ryan953**

### CODE REVIEW

File: static/app/utils/replays/resourceFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export function getResponseBodySize(frame: SpanFrame) {
+  if (isRequestFrame(frame)) {
+    // `data.responseBodySize` is from SDK version 7.44-7.45
+    return frame.data.response?.size ?? frame.data.responseBodySize;
+  }
+  if (isResourceFrame(frame)) {
+    // What about these?
+    //   frame.data.decodedBodySize
+    //   frame.data.encodedBodySize
+    return frame.data.size;
```
Comment (billyvg) [inline static/app/utils/replays/resourceFrame.tsx:39]: There's also transferred size -- using these 3 sizes we could deduce 1) compression and 2) caching, but would need some slight design inspo here

File: static/app/views/replays/detail/network/details/getOutputType.tsx [hunk 2]
```diff
@@ -16,41 +17,42 @@ type Args = {
...
   if (isReqUrlSkipped || isRespUrlSkipped) {
     return Output.URL_SKIPPED;
   }
 
   if (['request', 'response'].includes(visibleTab)) {
-    const isReqBodySkipped = reqWarnings?.includes('BODY_SKIPPED');
-    const isRespBodySkipped = respWarnings?.includes('BODY_SKIPPED');
+    // @ts-expect-error: is BODY_SKIPPED really emitted from the SDK?
+    const isReqBodySkipped = reqWarnings.includes('BODY_SKIPPED');
+    // @ts-expect-error: is BODY_SKIPPED really emitted from the SDK?
```
Comment (billyvg) [inline static/app/views/replays/detail/network/details/getOutputType.tsx:54]: Hmm yeah I don't see this in the SDK

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/resourceFrame.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/utils/replays/resourceFrame.tsx:39]: Ticket for later: https://github.com/getsentry/sentry/issues/53459

File: static/app/views/replays/detail/network/details/getOutputType.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/replays/detail/network/details/getOutputType.tsx:54]: To followup: https://github.com/orgs/getsentry/projects/23/views/3?pane=issue&itemId=34084925

**Pull request merged**